cpp/daal/src/data_management/data_conversion.cpp

```diff
@@ -152,12 +152,14 @@ DAAL_EXPORT vectorCopy2vFuncType getVector<int>()
 
 DAAL_EXPORT vectorConvertFuncType getVectorUpCast(int idx1, int idx2)
 {
+    DAAL_ASSERT(idx1 < 10);
     static vectorConvertFuncType table[][3] = DAAL_CONVERT_UP_TABLE(vectorConvertFunc);
     return table[idx1][idx2];
 }
 
 DAAL_EXPORT vectorConvertFuncType getVectorDownCast(int idx1, int idx2)
 {
+    DAAL_ASSERT(idx1 < 10);
     static vectorConvertFuncType table[][3] = DAAL_CONVERT_DOWN_TABLE(vectorConvertFunc);
```

Comment:
Please consider using [this technique](https://stackoverflow.com/questions/10274162/how-to-find-2d-array-size-in-c) instead